src/SixLabors.Fonts/Tables/AdvancedTypographic/UnicodeScriptTagMap.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using SixLabors.Fonts.Unicode;
+
+namespace SixLabors.Fonts.Tables.AdvancedTypographic
+{
+    /// <summary>
+    /// Provides a map from Unicode <see cref="ScriptClass"/> to OTF <see cref="Tag"/>.
+    /// <see href="https://docs.microsoft.com/en-us/typography/opentype/spec/scripttags"/>
+    /// </summary>
+    internal sealed class UnicodeScriptTagMap : Dictionary<ScriptClass, Tag[]>
```

Comment:
We use `TryGet` I believe so not currently.

Reply:
It's easy to implement `TryGet`: We can store `null` in the array for values which are not in `UnicodeScriptTagMap.Instance` but for some reason defined in `ScriptClass`.

Also note that all current values are arrays of `Length == 1`, so maybe we can store `Tag?` instead of `Tag[]`?

Reply:
Note that I don't expect all these nits to be resolved in the PR, but would be nice to add TODO notes or a tracking issue documenting the code quality / perf leftovers.

Reply:
I recall why this is an array now. Historically there's been changes to features over the years with some made obsolete that we still need to associate with a script tag. There's evidence of various substitutions scattered throughout the HarfBuzz code but I haven't gotten around to collating them.

Reply:
Turns out I already had a TODO for that in the code! 😄